### ISSUE TITLE
Add HTML content option for sections config

### ISSUE THREAD

**Issue by lukenofurther** [vue-styleguidist/vue-styleguidist#304]
My use case is this - source the content for pages/sections from an API rather than markdown files. In order to achieve this I'm writing HTML from the API into markdown files then feeding them into vue-styleguidist. Having the option of specifying an HTML string for content directly in the sections config instead of a markdown file path makes much more sense in this scenario.

**Comment by elevatebart** [comment]
That is a very interresting workflow indeed. I would love to hear more.
Are you saying that you build webpages using another tool, then pass them to styleguidist to build a documentation website ?

**Comment by ljelewis** [comment]
The idea is for page content to be curated in a CMS to allow non-devs to update it without having to ask devs to change the source code. Most CMS's these days have an API to retrieve the content, so the build fetches the page content and incorporates it into the styleguide instead of static markdown files. 

At the moment I'm doing this for the static (i.e. non-component) pages defined in the sections key of the config as part of a fork of vue-design-system. The workflow I've come up with is quite a fudge because vue-styleguidist is designed to use markdown files for it's documentation:

1. Poll CMS API for rich-text content
2. Convert rich-text to markdown/HTML
3. Write to .md file
4. Edit the sections config to incorporate the CMS content in the .md files
5. Trigger the styleguide build as normal with the edited config
6. Remove the generated markdown files

Being able to specify section content (and perhaps component content too) as Markdown, AST and/or HTML would simplify this workflow by removing steps 3 and 6.

I'd like to extend this to component documentation as well as static sections. I haven't looked at this task in any detail yet but it might be possible to leverage the getExampleFilename config option.

**Comment by elevatebart** [comment]
Thank you lj I am now intrigued ant want to participate !! You got me hooked.

**Comment by JessicaSachs** [comment]
Literally was just talking to @elevatebart about this
I want this. We're using Ghost CMS at work for our designers... 

@elevatebart you have that plugin architecture ready for vue-styleguidist yet?

**Comment by ljelewis** [comment]
I'm using Contentful and currently have the sections config key building off it. I also added the title key and some others might be useful. I have tokens in mind too eventually. For now I'll copy my changes over to a fork of vue-styleguidist to share. My solution utilises the contentful SDK so couldn't be used for other remote sources without changes.

@JessicaSachs have you done a similar integration for Ghost?

**Comment by JessicaSachs** [comment]
@ljelewis no, I haven't. Paste a link to what you did and I'll give it a go ^.^

**Comment by lukenofurther** [comment]
@elevatebart @JessicaSachs I've incorporated my changes into my fork of vue-styleguidist. Have a look at the remote-sections example I've added:

https://github.com/lukenofurther/vue-styleguidist

This is obviously just a POC at this stage and is very disruptive to the API of the main script because it converts it into an asynchronous function. I'd love to hear what you think.

**Comment by elevatebart** [comment]
Hey @lukenofurther !!! 

I believe it's finally time we get into this feature. 
I am planning on adding a lot of `async` in the mix here. 
The addition of Typescript will sure help.

I finally got around to look at your branch and it might be pretty awesome. Opens tons of doors.

Thank again for your patience.
target: end of September.

**Comment by lukenofurther** [comment]
Hi @elevatebart, that's great news!

Since I last shared I've changed my build so it doesn't rely on async within vue-styleguidist itself. The script now handles all the async before calling vue-styleguidist with the merged config object.

Still, adding async functionality into vue-styleguidist would make this easier for people to accomplish, so that's good.

There are some important considerations with the API call to allow for any API to be used. Authentication obviously, as well as transforming the response as responses from CMS APIs are very different from one another. 

Let me know if you want to see how I've handled that and/or any other way I can help.

And once again, great you found the time for this feature 👍
